File: backend/src/main/java/com/dev/backend/entity/Author.java
package com.dev.backend.entity;

import java.util.List;

import com.dev.backend.constant.BaseStatus;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "authors")
public class Author {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private String slug;
    private String wikibaseItem;
    @Column(columnDefinition = "TEXT")
    private String urlImage;
    private String urlBio;
    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private BaseStatus status = BaseStatus.ACTIVE;

    @OneToMany(mappedBy = "author")
    private List<ProductAuthor> productAuthors;
}

File: frontend/src/types/author.ts
import type { BaseStatus } from "./status";


export interface AuthorResponse {
  id: number;
  name: string;
  wikibaseItem: string;
  urlImage: string;
  description: string;
  urlBio: string;
  status: BaseStatus;
}
export interface AuthorRequest {
  name: string;
  wikibaseItem: string;
  urlImage: string;
  urlBio: string;
  extract: string;
  status: BaseStatus;
}
